server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -431,6 +431,7 @@ public void doRun()
                   mergedFile,
                   sink.getSegment().withDimensions(Lists.newArrayList(index.getAvailableDimensions()))
               );
+              index.close();
```

Comment:
It doesn't seem that creating index is even needed here, `IndexMerger.getMergedDimensions(indexes)` could be used